app/javascript/components/enrollment/steps/Exposure.js

```diff
@@ -235,6 +241,7 @@ class Exposure extends React.Component {
             }
 
             if (await confirmDialog(message, options)) {
+              self.setState({ selected_jurisdiction: self.state.current.patient.jurisdiction_id });
```

Comment:
So there is a bit of an issue with this approach where the `selected_jurisdiction` is not reset and it doesn't show the confirmation dialog when it should. 

To replicate:

1. Enroll a monitoree and change jurisdiction from USA -> USA, State 1 (for example, if you're a USA user)
2. Click next and see confirmation modal (correct)
3. Click previous, change jurisdiction back to USA and click next again and see no confirmation modal (correct)
4. Click previous again, change to USA -> USA, State 1 again and click next and notice there is NO confirmation modal (incorrect)

The problem is that the jurisdiction is changed back to the current value, `selected_jurisdiction` is not updated. You'll probably need to add a specific check for this case.

Reply:
Good catch, should be fixed with the latest commit